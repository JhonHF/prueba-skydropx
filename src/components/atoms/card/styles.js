import styled from "styled-components";

export const CardWrapper = styled.div`
  cursor: pointer;
  ${({ selected }) => selected && `border: 3px solid #4e34e1;`}
`;
