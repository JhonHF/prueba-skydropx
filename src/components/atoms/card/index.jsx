import React from "react";
import { CardWrapper } from "./styles";

export const Card = ({ children, selected, className, ...props }) => (
  <CardWrapper
    className={`card ${className || ""}`}
    selected={selected}
    id="card"
    {...props}
  >
    <div className="card-body">{children}</div>
  </CardWrapper>
);
