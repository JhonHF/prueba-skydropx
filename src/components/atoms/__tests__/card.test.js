import React from "react";
import { shallow } from "enzyme";

import { Card } from "../card";

describe("Card component", () => {
  describe("GIVEN a set of valid props", () => {
    const testProps = {
      className: "p-3",
      children: <p>Children</p>,
      selected: false,
      onClick: jest.fn(),
    };

    describe("WHEN the component is rendered without props", () => {
      const shallowWrapper = shallow(<Card {...testProps} />);

      it("THEN should display a regular AddPrestation", () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe("WHEN the card is selected", () => {
      const customProps = {
        ...testProps,
        selected: true,
      };

      const shallowWrapper = shallow(<Card {...customProps} />);

      it("THEN the onClick function is called", () => {
        expect(shallowWrapper.find("p").html()).toEqual("<p>Children</p>");
      });
    });
  });
});
