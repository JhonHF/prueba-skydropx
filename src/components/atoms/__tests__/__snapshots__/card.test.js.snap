// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component GIVEN a set of valid props WHEN the component is rendered without props THEN should display a regular AddPrestation 1`] = `
<styled.div
  className="card p-3"
  id="card"
  onClick={[MockFunction]}
  selected={false}
>
  <div
    className="card-body"
  >
    <p>
      Children
    </p>
  </div>
</styled.div>
`;
